include/OpenSpaceToolkit/Astrodynamics/NumericalSolver.hpp

```diff
@@ -230,14 +259,35 @@ class NumericalSolver
     /// @param              [in] aDurationArray An array of durations over which to integrate
     /// @param              [in] aSystemOfEquations A std::function wrapper with a particular signature that
     ///                              boost::odeint accepts to perform numerical integration
-    /// @return             std::vector<double>
+    /// @return             Array<Solution>
 
     Array<Solution> integrateDuration(
         const StateVector& anInitialStateVector,
         const Array<Real>& aDurationArray,
         const SystemOfEquationsWrapper& aSystemOfEquations
     );
 
+    /// @brief              Perform numerical integration from a start time till a condition is met, or end time is
+    /// reached
+    ///
+    /// @code
+    ///                     StateVector stateVector = numericalsolver.integrateDuration(stateVector,
+    ///                     aStartTime, aDurationInSeconds, SystemofEquations, anEventCondition) ;
+    /// @endcode
+    /// @param              [in] anInitialStateVector An initial n-dimensional state vector to begin integrating at
+    /// @param              [in] aDurationInSeconds A duration to integrate for
+    /// @param              [in] aSystemOfEquations An std::function wrapper with a particular signature that
+    ///                     boost::odeint accepts to perform numerical integration
+    /// @param              [in] anEventCondition An event condition
+    /// @return             Solution
+
+    Solution integrateDuration(
+        const StateVector& anInitialStateVector,
```

Comment:
Should we augment this to take into account `aStartTime`?

Reply:
So I understand this is not a big deal for now for us but just thinking about somebody using `integrateDuration` directly with a time dependent set of equations (anyway, feel free to ignore, this is not the purpose of this PR)